docs/reference/high-availability/cluster-design.asciidoc

```diff
@@ -230,24 +230,47 @@ The cluster will be resilient to the loss of any node as long as:
 [[high-availability-cluster-design-large-clusters]]
 === Resilience in larger clusters
 
-It is not unusual for nodes to share some common infrastructure, such as a power
-supply or network router. If so, you should plan for the failure of this
+It is not unusual for nodes to share some common infrastructure, such as network
+interconnects or a power supply. If so, you should plan for the failure of this
 infrastructure and ensure that such a failure would not affect too many of your
 nodes. It is common practice to group all the nodes sharing some infrastructure
 into _zones_ and to plan for the failure of any whole zone at once.
 
-Your cluster’s zones should all be contained within a single data centre. {es}
-expects its node-to-node connections to be reliable and have low latency and
-high bandwidth. Connections between data centres typically do not meet these
-expectations. Although {es} will behave correctly on an unreliable or slow
-network, it will not necessarily behave optimally. It may take a considerable
-length of time for a cluster to fully recover from a network partition since it
-must resynchronize any missing data and rebalance the cluster once the
-partition heals. If you want your data to be available in multiple data centres,
-deploy a separate cluster in each data centre and use
-<<modules-cross-cluster-search,{ccs}>> or <<xpack-ccr,{ccr}>> to link the
+{es} expects its node-to-node connections to be reliable and to have low
+latency and adequate bandwidth. Many of the tasks that {es} performs require
+multiple round-trips between nodes. This means that a slow or unreliable
+interconnect may have a significant effect on the performance and stability of
+your cluster. A few milliseconds of latency added to each round-trip can
+quickly accumulate into a noticeable performance penalty. {es} will
+automatically recover from a network partition as quickly as it can but your
+cluster may be partly unavailable during a partition and will need to spend
+time and resources to resynchronize any missing data and rebalance itself once
+the partition heals. Recovering from a failure may involve copying a large
```

Comment:
It seems like we're talking around reallocation and shard recovery here. Is there a reason we don't just directly mention and xref those two concepts?

Reply:
I didn't think we had any docs on those topics, at least not concept-level ones that would be suitable for linking from here. If you have some in mind then sure we can add links.

Reply:
I think you're right. We have some setting reference:

- https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-cluster.html
- https://www.elastic.co/guide/en/elasticsearch/reference/master/recovery.html

However, I don't think those are great links to use here. I've opened https://github.com/elastic/elasticsearch/issues/77515 to track this gap and add those docs.

This looks fine to me in the meantime.